src/sinks/datadog/metrics/encoder.rs

```diff
@@ -428,11 +414,17 @@ fn encode_timestamp(timestamp: Option<DateTime<Utc>>) -> i64 {
 fn generate_series_metrics(
     metric: &Metric,
     default_namespace: &Option<Arc<str>>,
+    log_schema: &'static LogSchema,
     last_sent: Option<Instant>,
 ) -> Result<Vec<DatadogSeriesMetric>, EncoderError> {
     let name = get_namespaced_name(metric, default_namespace);
+
+    let mut tags = metric.tags().cloned().unwrap_or_default();
+    let host = tags.remove(log_schema.host_key());
+    let source_type_name = tags.remove("source_type_name");
+    let device = tags.remove("device");
```

Comment:
We take it by reference because encoding is fallible.

If there was an error with the currently encoded metrics -- say the payload ended up being too big when we finalized it at the very end -- we want to be able to return all the metrics that were used to build the payload so that the caller has a chance to split them into smaller subbatches to hopefully encode them successfully.

You can see this around line 233 where we store the metrics to (potentially) be passed back around line 375 in `finish`.

Reply:
Hmm, I thought I had read down to the bottom of that function, but I missed that.